dataactbroker/handlers/accountHandler.py

```diff
@@ -137,6 +144,117 @@ def login(self,session):
+                                 "contact DATABroker@fiscal.treasury.gov to obtain access.")
+
+            cgac_group = list(filter(lambda g: len(g) == len(parent_group + "-CGAC_")+3, group_list))
+
+            # Deny access if they are not aligned with an agency
+            if not cgac_group:
+                raise ValueError("You have logged in with MAX but do not have permission to access the broker. Please "
+                                 "contact DATABroker@fiscal.treasury.gov to obtain access.")
+
+            try:
+                sess = GlobalDB.db().session
+                user = sess.query(User).filter(func.lower(User.email) == func.lower(email)).one_or_none()
+
+                # If the user does not exist, create them since they are allowed to access the site because they got
+                # past the above group membership checks
+                if user is None:
+                    user = User()
+
+                    first_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:First-Name']
+                    middle_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:Middle-Name']
+                    last_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:Last-Name']
+
+                    user.email = email
+
+                    # Check for None first so the condition can short-circuit without
+                    # having to worry about calling strip() on a None object
+                    if middle_name is None or middle_name.strip() == '':
+                        user.name = first_name + " " + last_name
+                    else:
+                        user.name = first_name + " " + middle_name[0] + ". " + last_name
+                    user.user_status_id = UserHandler().getUserStatusId('approved')
+
+                    # If part of the SYS agency, use that as the cgac otherwise use the first agency provided
+                    if list(filter(lambda g: g.endswith("SYS"), cgac_group)):
+                        user.cgac_code = "SYS"
+                        # website admin permissions
+                        user.permissions = 2
+                    else:
+                        user.cgac_code = cgac_group[0][-3:]
+                        # regular user permissions
+                        user.permissions = 1
+
+                    sess.add(user)
+                    sess.commit()
+
+            except MultipleResultsFound:
+                raise ValueError("An error occurred during login.")
+
+            # Create session
+            LoginSession.login(session, user.user_id)
+            permissionList = []
+            for permission in self.interfaces.userDb.getPermissionList():
```

Comment:
Again, my fault for not sharing this info more widely. Instead of using the `getPermissionList()` to get a list of permission, you can do `from dataactcore.models.lookups import PERMISSION_TYPE_DICT` to access a dictionary of permission type ids and values.